### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by weiznich**
This PR adds documentation to the crate. It also removes any item that does not need to be public from the public API. In addition it fixes some issues with how the dependencies are specified and remove the default feature set.

I propose to do a first release after merging this PR

### CODE REVIEW

File: src/oracle/connection/mod.rs [hunk 1]
```diff
@@ -347,8 +459,8 @@ impl OciConnection {
                     OciDataType::BigInt => oracle::sql_type::OracleType::Number(19, 0),
                     OciDataType::Float => oracle::sql_type::OracleType::Number(19, 0),
                     OciDataType::Double => oracle::sql_type::OracleType::BinaryDouble,
-                    OciDataType::Text => oracle::sql_type::OracleType::NVarchar2(2_000_000),
-                    OciDataType::Binary => oracle::sql_type::OracleType::Raw(2_000_000),
+                    OciDataType::Text => oracle::sql_type::OracleType::Varchar2(2000),
```
Comment (pgab) [inline src/oracle/connection/mod.rs:462]: Not sure, why, we change the type and size here?

File: src/oracle/connection/mod.rs [hunk 2]
```diff
@@ -360,8 +472,8 @@ impl OciConnection {
         for (n, b) in &other_binds {
             binds.push((n, &*b));
         }
-
-        stmt.execute_named(&binds).map_err(ErrorHelper::from)?;
+        //        dbg!(stmt.oci_attr::<oracle::oci_attr::Statement>());
```
Comment (pgab) [inline src/oracle/connection/mod.rs:475]: Is this some leftover?

File: src/oracle/connection/mod.rs [hunk 3]
```diff
@@ -360,8 +472,8 @@ impl OciConnection {
...
+        dbg!(stmt.execute_named(&binds)).map_err(ErrorHelper::from)?;
```
Comment (pgab) [inline src/oracle/connection/mod.rs:476]: Do we want `dbg!` here at all times?

File: src/test/mod.rs [hunk 4]
```diff
@@ -2394,5 +2394,72 @@ fn batch_insert_1() {
...
+    assert_eq!(res.2, Some("bac".into()));
+
+    let res2 = gst_types::table.get_result::<GSTTypes>(&mut conn).unwrap();
+    assert_eq!(res.0, res2.text);
+    assert_eq!(res.1, res2.byte);
+    assert_eq!(res.2, res2.v);
+}
+
+// #[test]
+// fn explicit_single_returning() {
```
Comment (pgab) [inline src/test/mod.rs:2437]: Do we want this test or not? If the first, it should be included, otherwise remove it completely.

**Review by pgab — CHANGES_REQUESTED** [review]

File: src/oracle/connection/mod.rs (hunk 2 above)
Comment (MrMuetze) [inline src/oracle/connection/mod.rs:475]: Can this be removed?

File: src/test/mod.rs [hunk 5]
```diff
@@ -2394,5 +2394,72 @@ fn batch_insert_1() {
     assert_eq!(res[1].big, Some(-3));
 }
 
+#[test]
```
Comment (MrMuetze) [inline src/test/mod.rs:2397]: Are both of these tests supposed to stay like that?

**Review by MrMuetze — CHANGES_REQUESTED** [review]
Just two questions remain. All the comments and docs look good!

**Comment by weiznich** [comment]
All of what was commented above was not supposed to be in this PR at all. Seems like I added that in the last commit that was only supposed to fix the CI. I've just rewrote the last commit so that's all excluded now.

**Review by pgab — APPROVED** [review]

**Review by MrMuetze — APPROVED** [review]

**Pull request merged**